cli_tools/gce_windows_upgrade/upgrader/workflows.go

```diff
@@ -524,7 +528,8 @@ func (u *upgrader) runWorkflowWithSteps(workflowName string, timeout string, pop
 		},
 	}
 
-	return w, daisyutils.NewDaisyWorker(w, env, u.logger).Run(map[string]string{})
+	err := daisyutils.NewDaisyWorker(workflowProvider, env, u.logger).Run(map[string]string{})
+	return wf, err
```

Comment:
I find the flow around `wf` convoluted: `NewDaisyWorker` will use `workflowProvider` to build a workflow but `wf` is modified by `workflowProvider` closure and this relies on the fact `NewDaisyWorker` will be called. 

Can daisyWorker store the workflow that was built to avoid this complexity?

Reply:
Thanks for the feedback.

> Can daisyWorker store the workflow that was built to avoid this complexity?

Unfortunately, no, since (as far as I know) a workflow can't be re-run.

Here's my logic: To retry the workflow using pd-standard disks, then we need a new workflow. Across the CLI tools, there's a myriad of customizations for creating workflows. The `workflowProvider` abstraction allows call sites to describe *how* to make the workflow, leaving `daisy_worker` to execute the workflow.

Do you think a different name would make it clearer? Or perhaps a different factoring?

Reply:
isn't the `wf` returned from here the same as `daisyWorker.finishedWf`?

Reply:
Got it, thanks! 

I looked some more, and the upgrade tool was exposing the workflow for unit testing. I updated the tests so that they inspected the worker, instead of the workflow.